code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm

```diff
@@ -118,56 +98,53 @@ Difficulty: Hard
 	if(.)
 		SSshuttle.shuttle_purchase_requirements_met |= "bubblegum"
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/do_attack_animation(atom/A, visual_effect_icon)
-	if(!charging)
-		..()
+/mob/living/simple_animal/hostile/megafauna/bubblegum/do_attack_animation(atom/A)
+	if(charging)
+		return
+	..()
 
 /mob/living/simple_animal/hostile/megafauna/bubblegum/AttackingTarget()
-	if(!charging)
-		return ..()
+	if(charging)
+		return
+	..()
 
 /mob/living/simple_animal/hostile/megafauna/bubblegum/Goto(target, delay, minimum_distance)
-	if(!charging)
-		..()
+	if(charging)
+		return
+	..()
 
 /mob/living/simple_animal/hostile/megafauna/bubblegum/Move()
+	if(!stat)
+		playsound(src.loc, 'sound/effects/meteorimpact.ogg', 200, 1, 2, 1)
 	if(charging)
-		new /obj/effect/temp_visual/decoy/fading(loc,src)
+		new/obj/effect/temp_visual/decoy/fading(loc,src)
 		DestroySurroundings()
 	. = ..()
-	if(!stat && .)
-		playsound(src, 'sound/effects/meteorimpact.ogg', 200, 1, 2, 1)
 	if(charging)
 		DestroySurroundings()
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/warp_charge()
-	blood_warp()
+/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/triple_charge()
+	charge()
+	sleep(10)
+	charge()
+	sleep(10)
 	charge()
 
-/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/charge(bonus_charges)
+/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/charge()
 	var/turf/T = get_turf(target)
 	if(!T || T == loc)
 		return
-	new /obj/effect/temp_visual/dragon_swoop/bubblegum(T)
-	charging = TRUE
+	new /obj/effect/temp_visual/dragon_swoop(T)
+	charging = 1
 	DestroySurroundings()
 	walk(src, 0)
 	setDir(get_dir(src, T))
 	var/obj/effect/temp_visual/decoy/D = new /obj/effect/temp_visual/decoy(loc,src)
-	animate(D, alpha = 0, color = "#FF0000", transform = matrix()*2, time = 3)
-	sleep(3)
-	throw_at(T, get_dist(src, T), 1, src, 0, callback = CALLBACK(src, .charge_end, bonus_charges))
-
-/mob/living/simple_animal/hostile/megafauna/bubblegum/proc/charge_end(bonus_charges, list/effects_to_destroy)
-	charging = FALSE
-	try_bloodattack()
-	if(target)
-		if(bonus_charges)
-			bonus_charges--
-			charge(bonus_charges)
-		else
-			Goto(target, move_to_delay, minimum_distance)
-			SetRecoveryTime(MEGAFAUNA_DEFAULT_RECOVERY_TIME)
+	animate(D, alpha = 0, color = "#FF0000", transform = matrix()*2, time = 5)
+	sleep(5)
```

Comment:
Ironically, this makes charges slower and easier to avoid.

Reply:
I'm just gonna, like, update the latest code to work like old bubblegum and give that to you, because there are a lot of errors/unnecessary regressions in here.

Reply:
http://puu.sh/AzESX/1ade5ca229.dm